File: src/solver/wrapper_dotblas.c
/*
 * This file is part of FreeFem++.
 *
 * FreeFem++ is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Lesser General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * FreeFem++ is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with Foobar.  If not, see <http://www.gnu.org/licenses/>.
 */

typedef void doublecomplex;
typedef void complex;
typedef int integer;
#include <vecLib/cblas.h>

/* Double Complex */ void zdotc_ (doublecomplex *ret_val, integer *n,
                                  doublecomplex *zx, integer *incx, doublecomplex *zy, integer *incy) {
	cblas_zdotc_sub(*n, zx, *incx, zy, *incx, ret_val);
}

/* Double Complex */ void zdotu_ (doublecomplex *ret_val, integer *n,
                                  doublecomplex *zx, integer *incx, doublecomplex *zy, integer *incy) {
	cblas_zdotu_sub(*n, zx, *incx, zy, *incx, ret_val);
	return;
}	/* zdotu___ */

/* Complex */ void cdotc_ (complex *ret_val, integer *n, complex *cx,
                           integer *incx, complex *cy, integer *incy) {
	cblas_cdotc_sub(*n, cx, *incx, cy, *incx, ret_val);
}	/* cdotc___ */

/* Complex */ void cdotu_ (complex *ret_val, integer *n, complex *cx,
                           integer *incx, complex *cy, integer *incy) {
	cblas_cdotu_sub(*n, cx, *incx, cy, *incx, ret_val);
}	/* cdotu___ */

File: src/solver/wrapper_dotblas.f
c  a compile sans underscore   -fno-underscoring 
c  gfortran -fno-underscoring  -O3 -c wrapper_dotblas.f
c -------        
        double complex function zdotc_(n, zx, incx, zy, incy)
        double complex zx(*), zy(*), z
        integer n, incx, incy
        
        call cblas_zdotc_sub(%val(n), zx, %val(incx), zy, %val(incy), z)
c        print*,'cblas_zdotc_sub' 
        zdotc_ = z
        return
        end
        
        double complex function zdotu_(n, zx, incx, zy, incy)
        double complex zx(*), zy(*), z
        integer n, incx, incy
        
        call cblas_zdotu_sub(%val(n), zx, %val(incx), zy, %val(incy), z)
c        print*,'cblas_zdotu_sub' 
        zdotu_ = z
        return
        end
        
        complex function cdotc_(n, cx, incx, cy, incy)
        complex cx(*), cy(*), c
        integer n, incx, incy
        
        call cblas_cdotc_sub(%val(n), cx, %val(incx), cy, %val(incy), c)
c        print*,'cblas_cdotc_sub' 
        cdotc_ = c
        return
        end
        
        complex function cdotu_(n, cx, incx, cy, incy)
        complex cx(*), cy(*), c
        integer n, incx, incy
        
        call cblas_cdotu_sub(%val(n), cx, %val(incx), cy, %val(incy), c)
c        print*,'cblas_cdotu_sub' 
        cdotu_ = c
        return
        end
